tests/integration/utils2.go

```diff
@@ -417,6 +418,57 @@ func executeTestCase(
 	}
 }
 
+// getCollectionNames gets an ordered, unique set of collection names across all nodes
+// from the action set within the given test case.
+//
+// It preserves the order in which they are declared, and shares indexes across all nodes, so
+// if a second node adds a collection of a name that was previously declared in another node
+// the new node will respect the index originally assigned.  This allows collections to be
+// referenced across multiple nodes by a consistent, predictable index - allowing a single
+// action to target the same collection across multiple nodes.
+func getCollectionNames(testCase TestCase) []string {
+	nextIndex := 0
+	collectionIndexByName := map[string]int{}
+
+	for _, a := range testCase.Actions {
+		switch action := a.(type) {
+		case SchemaUpdate:
+			if action.ExpectedError != "" {
+				// If an error is expected then no collections should result from this action
+				continue
+			}
+
+			splitByType := strings.Split(action.Schema, "type ")
```

Comment:
This is an edge case but the following schema type will cause problems here.

```
type user_type {
  name: String
}
```

Reply:
Since this is in our test suite and we define the type names, I don't mind this too much. Not sure it's worth trying to be clever in this case.

Reply:
Cheers Guys, that was a silly mistake of mine :)

> Since this is in our test suite and we define the type names, I don't mind this too much. Not sure it's worth trying to be clever in this case.

I think I agree with you, but I'll document this limitation somewhere obvious so it doesn't cost us too much time if we ever feel the need to define a collection of such a name. 

- [ ] Doc limitation

Reply:
@nasdf Give me a shout if you are not happy with the documentation-only-for-now approach and I can tweak this post merge